src/gui/ScintillaEditor.cc

```diff
@@ -132,7 +132,7 @@ const boost::property_tree::ptree& EditorColorScheme::propertyTree() const
   return pt;
 }
 
-ScintillaEditor::ScintillaEditor(QWidget *parent) : EditorInterface(parent)
+ScintillaEditor::ScintillaEditor(QWidget *parent, MainWindow &mainWindow) : EditorInterface(parent), mainWindow(mainWindow)
```

Comment:
This feels too coupled. Could you make the connection between the MainWindows and the editor from outside the editor instead?